src/bun.js/api/bun/socket.zig

```diff
@@ -1038,18 +1038,8 @@ pub const Listener = struct {
             TLSSocket.dataSetCached(tls.getThisValue(globalObject), globalObject, default_data);
 
             tls.doConnect(connection, socket_context) catch {
-                handlers_ptr.unprotect();
-                socket_context.deinit(true);
-                handlers.vm.allocator.destroy(handlers_ptr);
-                handlers.promise.deinit();
-                bun.default_allocator.destroy(tls);
-                const err = JSC.SystemError{
-                    .message = bun.String.static("Failed to connect"),
-                    .syscall = bun.String.static("connect"),
-                    .code = if (port == null) bun.String.static("ENOENT") else bun.String.static("ECONNREFUSED"),
-                };
-                exception.* = err.toErrorInstance(globalObject).asObjectRef();
-                return .zero;
+                tls.handleConnectError(@intFromEnum(bun.C.SystemErrno.ECONNREFUSED));
```

Comment:
should we still error instantly with `ENOENT` for unix sockets?

Reply:
Yeah makes sense for a moment I miss read a comment and removed it.

Reply:
fixed on https://github.com/oven-sh/bun/pull/9651/commits/073e1ae165e83f1951f55e3d4c396be3d6a77ec2